File: C(++)/[2 sem] 2DWorldSimulator/code/src/worldObjects/plants/SowThistle.cpp
//
// Created by szymo on 4/18/2023.
//

#include ".headers/SowThistle.h"

#define SOW_INITIATIVE 0
#define SOW_STRENGTH 0
#define SOW_SYMBOL '&'

SowThistle::SowThistle(int x, int y, World *world) : Plant(x, y, world, SOW_STRENGTH, SOW_INITIATIVE, SOW_SYMBOL) {}

SowThistle::~SowThistle() = default;

void SowThistle::action()
{
    for (int i = 0; i<3; i++) { Plant::action(); }
}

std::string SowThistle::getSpecieName() const
{
    return "SowThistle";
}

File: Java/[2 sem] 2DWorldSimulator/code/SowThistle.java
public class SowThistle extends Plant {
    private static final int SOW_INITIATIVE = 0;
    private static final int SOW_STRENGTH = 0;
    private static final char SOW_SYMBOL = '&';

    public SowThistle(int x, int y, World world) {
        super(x, y, world, SOW_STRENGTH, SOW_INITIATIVE, SOW_SYMBOL, 0);
        this.spreadChance = 2;
    }

    @Override
    public void action() {
        for (int i = 0; i < 3; i++) {
            super.action();
        }
    }

    @Override
    public String getSpecieName() {
        return "SowThistle";
    }
}